src/unit-number/unit-number-input.tsx

```diff
@@ -308,6 +323,7 @@ export const UnitNumberInput = ({
                     onFocus={handleFocus}
                     onBlur={handleBlur}
                     onChange={handleChange}
+                    onKeyDown={handleKeyDown}
                     disabled={disabled}
                     readOnly={readOnly}
                     type="text"
```

Comment:
missing the `inputMode` otherwise the keyboard will have non-numeric input options